app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,37 @@
+class VotesController < ApplicationController
+
+  def create
+    if params[:answer_id]
+      @votable = Answer.find(params[:answer_id])
+    else
+      @votable = Question.find(params[:question_id])
+    end
+
+    if !@votable.voted.include?(current_user.id)
+      @vote = @votable.votes.build(value: params[:value], voter: current_user)
```

Comment:
It would probably be better to implement a 
```
def has_voted? user
  votes.where(voter: user).count > 0
end
```
on the votable and call that here. The way you have it here is loading the whole collection then checking in ruby.